### ISSUE TITLE
[bug](vscode-extensions) VSCode extension conflicts with preset-icons causing crash without error message

### ISSUE THREAD

**Issue by gwsbhqt** [unocss/unocss#621]
Please click to view larger image.

![动画](https://user-images.githubusercontent.com/16471930/154612064-1c369aa6-657e-4e97-92d0-68408a94285c.gif)

![image](https://user-images.githubusercontent.com/16471930/154613747-73850e20-a88b-4ea7-a314-5d5aa132e822.png)

![image](https://user-images.githubusercontent.com/16471930/154613807-21f8671e-8812-4bee-897c-2d4124768428.png)

I made sure to uninstall and reinstall node/vscode/vscode extension/node_module properly and cleanly, I made sure vscode was reloaded, I made sure to only enable the UnoCss extension.

There is no detailed error message in the vscode debugger and the unocss plugin output, so the specific problem cannot be located. Hopefully there is a way to locate the problem.

Environmental information
```
Lenovo R9000P
Windows 10 LTSC 2021 x64
node 16.14.0
vscode 1.64.2
pnpm 6.31.0
typescript 4.5.5
vite 2.8.3
unocss 0.24.4
unocss vscode extension 0.24.4
```

package.json
```json
{
    "dependencies": {
        "@douyinfe/semi-ui": "^2.4.1",
        "ahooks": "^3.1.9",
        "classnames": "^2.3.1",
        "dayjs": "^1.10.7",
        "jotai": "^1.6.0",
        "react": "^18.0.0-rc.0",
        "react-dom": "^18.0.0-rc.0",
        "react-router": "^6.2.1",
        "react-router-dom": "^6.2.1",
        "sanitize.css": "^13.0.0",
        "swr": "^1.2.1",
        "valtio": "^1.3.0",
        "wretch": "^1.7.9"
    },
    "devDependencies": {
        "@iconify-json/eos-icons": "^1.1.1",
        "@iconify-json/mdi": "^1.1.0",
        "@rollup/pluginutils": "^4.1.2",
        "@swc/core": "^1.2.141",
        "@swc/helpers": "^0.3.3",
        "@types/classnames": "^2.3.1",
        "@types/jscodeshift": "^0.11.3",
        "@types/node": "^17.0.18",
        "@types/react": "^17.0.39",
        "@types/react-dom": "^17.0.11",
        "@typescript-eslint/eslint-plugin": "^5.12.0",
        "@typescript-eslint/parser": "^5.12.0",
        "@vitejs/plugin-react": "^1.2.0",
        "eslint": "^8.9.0",
        "eslint-config-prettier": "^8.3.0",
        "eslint-import-resolver-typescript": "^2.5.0",
        "eslint-plugin-import": "^2.25.4",
        "eslint-plugin-prettier": "^4.0.0",
        "eslint-plugin-react": "^7.28.0",
        "eslint-plugin-react-hooks": "^4.3.0",
        "eslint-plugin-sort-everything": "^0.0.18",
        "exec-sh": "^0.4.0",
        "jscodeshift": "^0.13.1",
        "just-scripts": "^1.8.0",
        "just-task": "^1.4.2",
        "less": "^4.1.2",
        "postcss": "^8.4.6",
        "postcss-less": "^6.0.0",
        "prettier": "^2.5.1",
        "stylelint": "^14.5.1",
        "stylelint-config-prettier": "^9.0.3",
        "stylelint-config-standard": "^25.0.0",
        "stylelint-less": "^1.0.3",
        "stylelint-order": "^5.0.0",
        "stylelint-prettier": "^2.0.0",
        "ts-node": "^10.5.0",
        "typescript": "^4.5.5",
        "unocss": "^0.24.4",
        "vite": "^2.8.3",
        "vite-plugin-inspect": "^0.3.14"
    },
    "scripts": {
        "build": "just build",
        "clean": "just clean",
        "dev": "vite",
        "postinstall": "ncu -u",
        "preview": "vite preview --host"
    }
}
```

tsconfig.json
```json
{
    "compilerOptions": {
        "declaration": true,
        "esModuleInterop": true,
        "jsx": "react-jsx",
        "module": "esnext",
        "moduleResolution": "node",
        "noEmit": true,
        "outDir": "dist",
        "resolveJsonModule": true,
        "skipLibCheck": true,
        "strict": true,
        "target": "esnext"
    },
    "ts-node": {
        "swc": true,
        "transpileOnly": true
    }
}
```

vite.config.ts
```ts
import react from '@vitejs/plugin-react'
import unocss from 'unocss/vite'
import { defineConfig } from 'vite'
import inspect from 'vite-plugin-inspect'

import automatic from './src/plugins/vite-automatic'

export default defineConfig({
    base: './',
    build: { emptyOutDir: false },
    optimizeDeps: { include: ['classnames'] },
    plugins: [react(), inspect(), unocss(), automatic()]
})
```

unocss.config.ts
```ts
import '@unocss/core'

import presetMini from '@unocss/preset-mini'
import { colors } from '@unocss/preset-mini/colors'
import { defineConfig, presetAttributify, presetIcons, presetUno } from 'unocss'

export default defineConfig({
    presets: [
        presetUno(),
        presetMini(),
        presetAttributify(),
        presetIcons({
            collections: {
                eosicons: () => import('@iconify-json/eos-icons/icons.json').then(i => i.default),
                mdi: () => import('@iconify-json/mdi/icons.json').then(i => i.default)
            },
            extraProperties: {
                display: 'inline-block',
                'font-size': '1.335em',
                'vertical-align': 'text-bottom'
            }
        })
    ],
    rules: [
        [
            /^box-reflect(?:-?(-?\d+))?$/,
            ([, x = 0]) => ({
                '-webkit-box-reflect': `below ${x}px linear-gradient(transparent 30%, rgba(0,0,0,0.7))`
            })
        ],
        [
            /bg-size-(\d+)(?:-(\d+))?/,
            ([, x, y]) => ({
                'background-size': `${x}% ${y ? `${y}%` : ''}`
            })
        ],
        [
            /bg-gradient(?:-(-?\d+))?(?:-([\w-]+))/,
            ([, x = 0, y]) => ({
                'background-image': `linear-gradient(${[
                    `${x}deg`,
                    ...y
                        .split('-')
                        .filter(Boolean)
                        .map(x => {
                            const color = colors?.[x]
                            return typeof color === 'object' ? color['500'] : x
                        })
                ].join(', ')})`
            })
        ]
    ],
    theme: {
        animation: {
            keyframes: {
                bgswing: '{0%,100%{background-position:0% 0%;}50%{background-position:100% 100%;}}'
            },
            properties: {
                'fade-in': { 'animation-fill-mode': 'forwards', 'animation-iteration-count': 1 },
                'fade-out': { 'animation-fill-mode': 'forwards', 'animation-iteration-count': 1 }
            }
        }
    }
})
```

TSX file
```tsx
import { type FC } from 'react'

const Home: FC = () => {
    return (
        <div>
            <span>Home</span>
            <i className="m-2" />
            <i className="i-mdi-home" />
        </div>
    )
}

export default Home
```

**Comment by gwsbhqt** [comment]
@antfu @chu121su12 The problem persisted and bothered me, I switched to different notebooks and environments. Is there any way to locate what exactly is the problem?

**Comment by vipero07** [comment]
I had a similar issue in the past, turned out that updating VSCode fixed it for me.

**Comment by antfu** [comment]
Should be fix 0.24.7, please upgrade all the packages and the extension. Thanks
